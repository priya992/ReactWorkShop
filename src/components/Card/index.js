import React, {useState} from 'react';

const Card = (props) => {
  const [image, setImage] = useState(`media/${props.value.['poster-image']}`)
  const onError = () => {
    setImage(`media/placeholder_for_missing_posters.png`)
  }

  return (
    <>
      <img
        src={image}
        alt="test"
        className="image-style"
        onError={onError}
      />
      <div className="card-text">{props.value.name}</div>
    </>
  )
}

export default Card;
