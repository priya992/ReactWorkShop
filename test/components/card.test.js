import React from 'react';
import renderer from 'react-test-renderer';

import Card from './../../src/components/Card';
import { cardProps } from './../__mock__';

test('Card component snapshot', () => {
  const component = renderer.create(
    <Card value={cardProps}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
