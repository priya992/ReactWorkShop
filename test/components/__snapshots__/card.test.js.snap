// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component 1`] = `
Array [
  <img
    alt="test"
    className="image-style"
    onError={[Function]}
    src="media/undefined"
  />,
  <div
    className="card-text"
  />,
]
`;
